include/SDL_mixer.h

```diff
@@ -2408,6 +2409,10 @@ extern DECLSPEC int SDLCALL Mix_PausedMusic(void);
  */
 extern DECLSPEC int SDLCALL Mix_ModMusicJumpToOrder(int order);
 
+/* Tracks */
+extern DECLSPEC void SDLCALL Mix_StartTrack(int track);
```

Comment:
How does this interact with starting and stopping the music playback? Does this start playing or does this just select the track to play? Should this maybe be `Mix_SelectTrack()` to more clearly show what is happening?

Reply:
Currently it only changes the track. If the music isn't playing already, it doesn't do anything. SelectTrack sounds like a better name.

Reply:
* Nothing happens if `Play` is not called before
* Otherwise, sets and plays the track. I think GME will restart a track if already playing

Not certain if point one is necessary. GME_StartTrack does call Play itself after all. Maybe the music_playing check should be removed in Mix_StartTrack

Reply:
@Wohlstand any thoughts on the API here?

Reply:
I am fully against making new API calls without the `Mix_Music*` pointer. This is a very bad idea, and it makes a lot of problems in various cases, like, you are required to play the music until you will are able to change the stuff. Since I made MixerX, all new music-related API calls always require the music object pointer, which will allow you to manipulate music objects without the necessary to play them. I do remind some users who reported related bugs to my MixerX side that they are unable to retrieve meta-data until they start playing the music object.

Reply:
I rather make the:
```c
extern DECLSPEC void SDLCALL Mix_SelectTrack(Mix_Music *music, int track);
```
instead.

Reply:
Also, while this intended to select the track before to play, this is also a way that should allow switching the track on the fly without recreating the music object and reloading the file.

> P.S. Off topic: I think, I could use the same API at my MixerX side to allow switching sub-melodies of XMI files on the fly (if you didn't know, XMI may contain many different melodies inside, however, my sequencer doesn't support the proper switching between tracks yet, that I working on)

Reply:
P.S. Forgot to tell, at my MixerX you may also find several music-related API calls that repeating exist calls of SDL_Mixer, but they have a music object pointer. They were intended to manipulate individual music objects without starting to play them. In addition, they majorly helped me to implement multi-music API which allows playing multiple independent music streams and manipulating each music object independently. So, that is why I am totally against making new music-related calls without providing the music object pointer to it.

Reply:
Agreed on the pointer thing.

But does GME's `gme_start_track` just switch the track or does it fail/do nothing if not already playing? Can't recall why I put a "Play" call in there.

Reply:
You may basically save the tracking number at the variable (at the internal structure) when music is inactive, and when starting its play, re-use the value from that variable. When music is running, also call the track switch (using that gme_start_track call).

Reply:
Turns out gme does that for us.